src/clientdlg.cpp

```diff
@@ -730,15 +719,12 @@ void CClientDlg::OnConnectDlgAccepted()
             }
         }
 
-        // first check if we are already connected, if this is the case we have to
-        // disconnect the old server first
-        if ( pClient->IsRunning() )
+        // initiate connection
+        if ( pClient->Connect ( strSelectedAddress, strMixerBoardLabel ) )
```

Comment:
This should be refactored imo. Feels very clumsy.

Reply:
UI should simply pass any captured details to Client and await feedback.  There shouldn't be any testing of the response.  (Any input validation should be factored out of Client if it needs sharing between Client and ClientDlg.)